firestore/document.go

```diff
@@ -299,23 +298,11 @@ func newDocumentSnapshot(ref *DocumentRef, proto *pb.Document, c *Client, readTi
 		proto: proto,
 	}
 	if proto != nil {
-		ts, err := ptypes.Timestamp(proto.CreateTime)
-		if err != nil {
-			return nil, err
-		}
-		d.CreateTime = ts
-		ts, err = ptypes.Timestamp(proto.UpdateTime)
-		if err != nil {
-			return nil, err
-		}
-		d.UpdateTime = ts
+		d.CreateTime = proto.CreateTime.AsTime()
+		d.UpdateTime = proto.UpdateTime.AsTime()
```

Comment:
I think `AsTime` will "just work" if the `*Timestamp` is `nil` based on [the implementation](https://github.com/protocolbuffers/protobuf-go/blob/v1.33.0/types/known/timestamppb/timestamp.pb.go#L199-L201) (which uses accessors for the fields so that they can return `0` if the receiver/proto is `nil`), but we really ought to be using accessors for proto fields e.g. `proto.GetCreateTime().AsTime()`.

Do you feel comfortable making those changes here? 

There are a few other examples of this as well and I won't comment on each until we've discussed.

Reply:
I can at least update for everywhere I have touched. Switching to accessors in general might be a larger project that likely greatly affects all veneers

Reply:
that's fair! I think that's right considering "type" of access is changing with your updates (the field is being "dereferenced" directly instead of just passed to a function that may handle the `nil` case)